Library/Homebrew/readall.rb

```diff
@@ -35,28 +36,43 @@ def valid_aliases?(alias_dir, formula_dir)
     end
 
     def valid_formulae?(formulae)
-      failed = false
+      success = true
       formulae.each do |file|
         Formulary.factory(file)
       rescue Interrupt
         raise
       rescue Exception => e # rubocop:disable Lint/RescueException
         onoe "Invalid formula: #{file}"
-        puts e
-        failed = true
+        $stderr.puts e
+        success = false
       end
-      !failed
+      success
+    end
+
+    def valid_casks?(casks)
```

Comment:
Could add a `os/linux` implementation that makes this return `true` unconditionally